code/datums/crew_objective.dm

```diff
@@ -706,63 +832,85 @@ ABSTRACT_TYPE(/datum/objective/crew/geneticist)
 /datum/objective/crew/geneticist/scanned
 	explanation_text = "Have at least 5 people's DNA scanned in the cloning console at the end of the round."
 	medal_name = "Life, uh... finds a way"
+	var/static/check_result = null
 	check_completion()
-		for(var/obj/machinery/computer/cloning/C as anything in machine_registry[MACHINES_CLONINGCONSOLES])
-			if(C.records.len > 4)
-				return 1
-		return 0
+		if(isnull(check_result))
+			check_result = FALSE
+			for(var/obj/machinery/computer/cloning/C as anything in machine_registry[MACHINES_CLONINGCONSOLES])
+				if(C.records.len > 4)
+					check_result = TRUE
+		return check_result
 
 /datum/objective/crew/geneticist/booth
 	explanation_text = "Have at least 5 options available in the gene booth at the end of the round."
+	var/static/check_result = null
 	check_completion()
-		var/list/geneoptions = list()
-		for_by_tcl(GB, /obj/machinery/genetics_booth)
-			geneoptions |= GB.offered_genes
-		return length(geneoptions) >= 5
+		if(isnull(check_result))
+			var/list/geneoptions = list()
+			for_by_tcl(GB, /obj/machinery/genetics_booth)
+				geneoptions |= GB.offered_genes
+			check_result = length(geneoptions) >= 5
+		return check_result
 
 ABSTRACT_TYPE(/datum/objective/crew/roboticist)
 /datum/objective/crew/roboticist/cyborgs
 	explanation_text = "Ensure that there are at least three living cyborgs at the end of the round."
 	medal_name = "Progenitor"
+	var/static/check_result = null
 	check_completion()
 		var/borgcount = 0
-		for(var/mob/living/silicon/robot in mobs) //borgs gib when they die so no need to check stat I think
-			borgcount ++
-		if(borgcount > 2) return 1
-		else return 0
+		if(isnull(check_result))
+			check_result = FALSE
+			for(var/mob/living/silicon/robot/robot in mobs) //borgs gib when they die so no need to check stat I think
+				borgcount ++
+			if(borgcount > 2)
+				check_result = TRUE
+		return check_result
 
 /datum/objective/crew/roboticist/medibots
 	explanation_text = "Have at least five medibots on the station level at the end of the round."
 	medal_name = "Silent Running"
+	var/static/check_result = null
 	check_completion()
 		var/medbots = 0
-		for (var/obj/machinery/bot/medbot/M in machine_registry[MACHINES_BOTS])
-			if (M.z == 1)
-				medbots++
-		if (medbots > 4) return 1
-		else return 0
+		if(isnull(check_result))
+			check_result = FALSE
```

Comment:
same for all the others: why not just have it start as false?

Reply:
I believe I left a comment about this... somewhere. Regardless, it starts as `null` as an indicator that the check hasn't been executed yet. When it has been executed, it'll no longer be null and instead TRUE or FALSE; the isnull() will then fail and the proc will immediately return the TRUE or FALSE static var.